icu4c/source/test/testdata/BUILDRULES.py

```diff
@@ -63,7 +63,7 @@ def generate_rb(config, io, common_vars):
             input_files = [InFile("%s.txt" % bn) for bn in basenames],
             output_files = [OutFile("%s.res" % bn) for bn in basenames],
             tool = IcuTool("genrb"),
-            args = "-q -s {IN_DIR} -d {OUT_DIR} {INPUT_FILE}",
+            args = "-q -i {OUT_DIR}/../../../../data/out/build -s {IN_DIR} -d {OUT_DIR} {INPUT_FILE}",
```

Comment:
This is probably going to be brittle. If we need to know the path to the non-test data files while running genrb for test data files, we should provide that path in the environment. Here's how you would plumb it through:

1. Pick a name for the variable; something like `LIBRARY_DATA_DIR`
2. Calculate the path in `icu4c/source/test/testdata/Makefile.in` and set it to the variable; see "Variable names for rules.mk" in that file
3. Edit `icu4c/source/python/icutools/databuilder/__main__.py` to plumb that variable into the `makefile_env`. I'm not sure about running the tool in non-Makefile mode; presumably you have the same problem, so you may need to add a flag to `databuilder` like we do for all the other paths.
4. Now you can use that variable in this args path: `{LIBRARY_DATA_DIR}`